Adding new swift protocol for issue #2962 (#3131)
Codecademy/docs: content/swift/concepts/protocols/terms/codingkeyrepresentable/codingkeyrepresentable.md

---
Title: 'Protocols'

> Description: 'A protocol type that gives you the ability to convert it into a coding key and vice versa.'

Reviewer: ```suggestion
Description: 'A protocol that dictates the use of a key type that can encode and decode from a KeyedContainer.'
```

Subjects:
  - 'iOS'
  - 'Mobile Development'
Tags:
  - 'Protocols'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> 

Reviewer: remove extra empty line

> The `CodingKeyRepresentable` protocol allows you to covert a CodingKey type between a personalized type. This protocol enables you to use dictionaries that are not a string or Int(non-standard keys) to be encoded using organized 'keyed' containers instead of 'unkeyed' containers.

Reviewer: ```suggestion
The `CodingKeyRepresentable` protocol provides the flexibility to covert between a `CodingKey` type between a personalized type. This protocol enables the use of dictionaries that are not a `String` or `Int` (non-standard keys) to be encoded using organized 'keyed' containers instead of 'unkeyed' containers.
```

Reviewer: avoid the use of first and second person pronouns (I you we)

## Syntax
```pseudo
protocol CodingKeyRepresentable
```

> 

Reviewer: Remove extra line. There should never be more than one empty line between content

> > **Note:** CodingKeyRepresentable lets you create methods to convert your data into a string format that works as a key. This is particularly useful when using your own custom structs as keys

Reviewer: ```suggestion
> **Note:** `CodingKeyRepresentable` allows for the creation of methods to convert data into a string format that function as a key. This is particularly useful when implementing custom structs as keys.
```

> 

Reviewer: ```suggestion
remove extra lines
```

> ## Example

Reviewer: add a line after the example header

> In the following example, a special list called an “enum” named `NewSettings` which has two options: “name” and “instagram”. This list helps to organize data(for e.g. settings for a profile). A dictionary named `newDict` is created, it uses the special list as keys and strings as values. Then a tool called “JSONEncoder” is used to convert this dictionary into a special format called JSON(which is often used to share data between programs). The JSON data is turned into readable text and the output is shown in an object format.

Reviewer: ```suggestion
In the following example, a special list, known as an enum, named `NewSettings` is declared. This enum has two options: `name` and `instagram`. This list can be used to organize data (e.g. settings for a profile). A dictionary named `newDict` is created, and it uses the special list as keys and strings as values. 

Then a tool called a `JSONEncoder` is used to convert this dictionary into a special format called [JSON](https://www.codecademy.com/resources/docs/general/json). The JSON data is turned into readable text and the output is shown in an object format.
```
